test/state/host.cpp

```diff
@@ -77,24 +77,35 @@ uint256be Host::get_balance(const address& addr) const noexcept
     return (acc != nullptr) ? intx::be::store<uint256be>(acc->balance) : uint256be{};
 }
 
+namespace
+{
+/// For EXTCODE* instructions if the target is an EOF account, then only return EF00.
+/// While we only do this if the caller is legacy, it is not a problem doing this
+/// unconditionally, because EOF contracts dot no have EXTCODE* instructions.
+bytes_view extcode(bytes_view code, evmc_revision rev) noexcept
+{
+    return (rev >= EVMC_PRAGUE && is_eof_container(code)) ? code.substr(0, 2) : code;
```

Comment:
Unless I'm missing something, the check `rev >= EVMC_PRAGUE` is not needed. The EOF prefix is picked in a way to guarantee that there is legacy code with such prefix in the state. In other words, the `is_eof_container(code)` implies `rev >= EVMC_PRAGUE`.

Maybe we should also drop the check in https://github.com/ethereum/evmone/blob/master/test/statetest/statetest_loader.cpp#L445.

Reply:
> there is legacy code with such prefix in the state
Presuming you've meant "no legacy code with such prefix", I see the point. I think the rule is too weak though. We can warn against activating EOF EIPs on networks which might have `0xEF00` legacy bytecodes (we only our networks don't), but they should function correctly if they don't activate EOF, i.e. pre-Prague.

This is maybe dogmatic, but seems safer, WDYT?

Reply:
Decided on the call to assume as suggested and remove the `rev` check. Also EIP-3540 will include a note that no non-EOF MAGIC prefixed code can be present on a chain which adopts the eip.